experimental/framework/tree-react-api/src/useTree.ts

```diff
@@ -0,0 +1,30 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Tree, type TreeNode } from "@fluidframework/tree";
+import * as React from "react";
+
+/**
+ * Custom hook which invalidates a React Component when there is a change in the subtree defined by `subtreeRoot`.
+ * This includes changes to the tree's content, but not changes to its parentage.
+ * See {@link @fluidframework/tree#TreeChangeEvents.treeChanged} for details.
+ * @privateRemarks
+ * Without a way to get invalidation callbacks for specific fields,
+ * its impractical to implement an ergonomic and efficient more fiend grained invalidation hook.
+ * @public
+ */
+export function useTree(subtreeRoot: TreeNode): void {
+	// Use a React effect hook to invalidate this component when the inventory changes.
+	// We do this by incrementing a counter, which is passed as a dependency to the effect hook.
+	const [invalidations, setInvalidations] = React.useState(0);
+
+	// React effect hook that increments the 'invalidation' counter whenever inventory or any of its children change.
+	React.useEffect(() => {
+		// Returns the cleanup function to be invoked when the component unmounts.
+		return Tree.on(subtreeRoot, "treeChanged", () => {
```

Comment:
I'm not following this logic. Why are we subscribing to `Tree.on` when the component unmounts? Shouldn't this subscribe when the effect is run, and unsubscribe in the cleanup function?

Reply:
we are unsubscribing from tree.on when it unmounts. I think you read it wrong.

Reply:
Oh, `Tree.on` returns a cleanup function. I see.